Makefile

```diff
@@ -1,47 +1,63 @@
-ifndef GOPATH
-$(error No GOPATH set)
-endif
+SHORT_NAME := helm
+DEIS_REGISTRY ?= ${DEV_REGISTRY}
+IMAGE_PREFIX ?= helm
+
+REPO_PATH := github.com/helm/${SHORT_NAME}
 
+# The following variables describe the containerized development environment
+# and other build options
 BIN_DIR := bin
 DIST_DIR := _dist
 GO_PACKAGES := action chart config dependency log manifest release plugins/sec plugins/example codec
 MAIN_GO := helm.go
 HELM_BIN := $(BIN_DIR)/helm
 PATH_WITH_HELM = PATH="$(shell pwd)/$(BIN_DIR):$(PATH)"
-
 VERSION := $(shell git describe --tags --abbrev=0 2>/dev/null)+$(shell git rev-parse --short HEAD)
+DEV_ENV_IMAGE := quay.io/deis/go-dev:0.9.0
+DEV_ENV_WORK_DIR := /go/src/${REPO_PATH}
+DEV_ENV_CMD := docker run --rm -v ${CURDIR}:${DEV_ENV_WORK_DIR} -w ${DEV_ENV_WORK_DIR} ${DEV_ENV_IMAGE}
+DEV_ENV_CMD_INT := docker run -it --rm -v ${CURDIR}:${DEV_ENV_WORK_DIR} -w ${DEV_ENV_WORK_DIR} ${DEV_ENV_IMAGE}
+LDFLAGS := "-s -X main.version=${VERSION}"
 
-export GO15VENDOREXPERIMENT=1
```

Comment:
This shouldn't go away.  It's already set inside the container, but there should be _one_ step like `binary-build` (which I think we should rename to `native-build`) that _might_ be executed outside the container if a developer really wants to build helm from source for their laptop's native OS and architecture _only_.  For the sake of that one edge case, this variable should remain.